huaweicloud/services/live/resource_huaweicloud_live_recording.go

```diff
@@ -0,0 +1,387 @@
+							Optional: true,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"hls": {
+				Type:         schema.TypeList,
+				Optional:     true,
+				MaxItems:     1,
+				AtLeastOneOf: []string{"hls", "flv", "mp4"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"recording_length": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(15, 720), // same with console
+						},
+
+						"file_naming": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+
+						"ts_file_naming": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+						},
+
+						"max_stream_pause_length": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(-1, 300),
+							Computed:     true,
+						},
+					},
+				},
+			},
+
+			"flv": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem:     formatSchema(),
+			},
+
+			"mp4": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem:     formatSchema(),
+			},
+
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice([]string{recordingTypeAuto, recordingTypeManual}, false),
+			},
+		},
+	}
+}
+
+func formatSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"recording_length": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(15, 180), // same with console
+			},
+
+			"file_naming": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"max_stream_pause_length": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ValidateFunc: validation.IntBetween(0, 300),
+				Computed:     true,
+			},
+		},
+	}
+}
+
+func resourceRecordingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcLiveV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Live v1 client: %s", err)
+	}
+
+	createBody, err := buildRecordingParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	createOpts := &model.CreateRecordRuleRequest{
+		Body: createBody,
+	}
+
+	log.Printf("[DEBUG] Create Live recording params : %#v", createOpts)
+
+	resp, err := client.CreateRecordRule(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Live recording: %s", err)
+	}
+
+	if resp.Id == nil {
+		return diag.Errorf("error creating Live recording: ID not found in the API response")
+	}
+
+	d.SetId(*resp.Id)
+
+	return resourceRecordingRead(ctx, d, meta)
+}
+
+func resourceRecordingRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
replace ctx with '_' as it's not used in this func

Reply:
done